h2/src/main/org/h2/mvstore/db/MVSecondaryIndex.java

```diff
@@ -228,9 +228,9 @@ public void add(Session session, Row row) {
     }
 
     private void requireUnique(SearchRow row, TransactionMap<Value, Value> map, ValueArray unique) {
-        Iterator<Value> it = map.keyIterator(unique);
-        if (it.hasNext()) {
-            ValueArray k = (ValueArray) it.next();
+        Value key = map.ceilingKey(unique);
```

Comment:
Wouldn't this call never return useful row (same index keys)? ValueArray has a ValueLong.MIN as a last element, so even if prospective duplicate entry (committed or otherwise) does exist, it would never be returned. The whole method become noop.

Reply:
No, it founds matches on attempts of unique index violations.

Reply:
You mean in unit tests? But how do you know? That very well might be a result of second check after row insertion.

Reply:
Unit tests with debugger.